ipaclient/plugins/vault.py

```diff
@@ -703,14 +703,39 @@ def _get_vaultconfig(self, force_refresh=False):
         return transport_cert, wrapping_algo
 
     def _do_internal(self, algo, transport_cert, raise_unexpected,
-                     *args, **options):
+                     use_oaep=False, *args, **options):
         public_key = transport_cert.public_key()
 
         # wrap session key with transport certificate
-        wrapped_session_key = public_key.encrypt(
-            algo.key,
-            padding.PKCS1v15()
-        )
+        # KRA may be configured using either the default PKCS1v15 or RSA-OAEP.
+        # there is no way to query this info using the REST interface.
+        if not use_oaep:
+            # PKCS1v15() causes an OpenSSL exception when FIPS is enabled
+            # if so, we fallback to RSA-OAEP
+            try:
+                wrapped_session_key = public_key.encrypt(
+                    algo.key,
+                    padding.PKCS1v15()
+                )
+            except ValueError:
```

Comment:
It will never get here because the session key is used later. This was my proposed fallback mechanism where the call the self._do_internal() was embedded in the try/except. It's ok that you moved the fallback to the end but I don't think we need this try/except now.

Reply:
yep, it feels somewhat duplicated, I added it for a specific reason:

- in FIPS mode, the PKCS1v15() immediately fails because openssl is preventing the use of the padding. Then, we see:

```
  File "/usr/lib64/python3.12/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 163, in _enc_dec_rsa_pkey_ctx
    raise ValueError("Encryption/decryption failed.")
ValueError: Encryption/decryption failed.
```

so that try/except is catching the openssl attempt and falling back into OAEP (that should work because in FIPS we default to OAEP).

btw, I wanted to keep _do_internal() this way so that we benefit from the transport cert cache.

Reply:
Got it